src/shared/pager.c

```diff
@@ -53,6 +53,11 @@ noreturn static void pager_fallback(void) {
         _exit(EXIT_SUCCESS);
 }
 
+static int stored_stdout = -1;
+static int stored_stderr = -1;
+static bool stdout_redirected = false;
+static bool stderr_redirected = false;
```

Comment:
Sure, but `stdout_redirected` is used for one thing and one thing only: to decide if `stored_stdout` should be moved back as stdout. But that's duplicate information: if `stored_stdout >= 0`, it's always subject to being moved back, and otherwise there's nothing to do.